File: Lab 2/Raveenth_100704540/Confluent Kafka Python/topic.py
from kafka.admin import KafkaAdminClient, NewTopic
import sys

try: 
    admin_client = KafkaAdminClient( 
        client_id='myapp',
        bootstrap_servers = [ 
            "localhost:9093",
            "localhost:9094",
            "localhost:9095"
        ]
    )

    topic_list = []
    topic_list.append(NewTopic(name="topic", num_partitions=2, replication_factor=2))
    admin_client.create_topics(new_topics=topic_list, validate_only=False)
    print("Topic has been created!")

except Exception as e:
    print("Topic could not be created: ", e.__class__)

finally:
    sys.exit()

File: Lab 2/Raveenth_100704540/Kafka NodeJS/topic.js
//const Kafka = require("kafkajs").Kafka
const {Kafka} = require("kafkajs")

run();

async function run(){
    try {
        const kafka = new Kafka({
            "clientId": "myapp",
            "brokers": [
                "localhost:9093",
                "localhost:9094",
                "localhost:9095"
            ]
        })

        const admin = kafka.admin();
        console.log("Connecting.......");
        await admin.connect()
        console.log("Connected!")
        await admin.createTopics({
            "topics" : [{
                "topic" : "topic",
                "numPartitions": 2
            }]
        })
        console.log("Created successfully")
    } 
    catch(ex)
    {
        console.error(`Something bad happened ${ex}`)
    }
    finally{
        process.exit(0);
    }
}